dbus/mako.c

```diff
@@ -71,18 +71,21 @@ static int handle_reload(sd_bus_message *msg, void *data,
 		sd_bus_error *ret_error) {
 	struct mako_state *state = data;
 
-	reload_config(&state->config);
+	if (!reload_config(&state->config)) {
+		return sd_bus_reply_method_return(msg, "b", false);
+	}
+
 	send_frame(state);
 
-	return sd_bus_reply_method_return(msg, "");
+	return sd_bus_reply_method_return(msg, "b", true);
 }
 
 static const sd_bus_vtable service_vtable[] = {
 	SD_BUS_VTABLE_START(0),
 	SD_BUS_METHOD("DismissAllNotifications", "", "", handle_dismiss_all_notifications, SD_BUS_VTABLE_UNPRIVILEGED),
 	SD_BUS_METHOD("DismissLastNotification", "", "", handle_dismiss_last_notification, SD_BUS_VTABLE_UNPRIVILEGED),
 	SD_BUS_METHOD("InvokeAction", "s", "", handle_invoke_action, SD_BUS_VTABLE_UNPRIVILEGED),
-	SD_BUS_METHOD("Reload", "", "", handle_reload, SD_BUS_VTABLE_UNPRIVILEGED),
+	SD_BUS_METHOD("Reload", "", "b", handle_reload, SD_BUS_VTABLE_UNPRIVILEGED),
```

Comment:
Would it be better to use dbus errors instead of return values?

Reply:
I didn't know those existed, but probably!